### PR TITLE
blocked: import go-ipld-cbor interface

### PR DESCRIPTION
**Pull request by frrist**
Blocked on ipfs/go-ipld-cbor#70

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 	github.com/spaolacci/murmur3 v1.1.0
-	github.com/whyrusleeping/cbor-gen v0.0.0-20190910031516-c1cbffdb01bb
-	golang.org/x/xerrors v0.0.0-20190717185122-a985d3407aa7
+	github.com/warpfork/go-wish v0.0.0-20200122115046-b9ea61034e4a // indirect
+	github.com/whyrusleeping/cbor-gen v0.0.0-20200123233031-1cdf64d27158
+	golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543
 )
 
+replace github.com/ipfs/go-ipld-cbor => ../go-ipld-cbor
+
```
Comment (frrist) [inline go.mod]: will remove before merging

File: ipld.go [hunk 2]
```diff
@@ -1,197 +0,0 @@
-package hamt
```
Comment (frrist) [inline ipld.go]: moved to https://github.com/ipfs/go-ipld-cbor/pull/70/files#diff-1ac94134b8964626aaac00790fa9b48f

File: go.mod [hunk 3]
```diff
@@ -4,11 +4,13 @@ require (
 	github.com/ipfs/go-block-format v0.0.2
 	github.com/ipfs/go-cid v0.0.3
 	github.com/ipfs/go-ipld-cbor v0.0.3
-	github.com/multiformats/go-multihash v0.0.8
-	github.com/polydawn/refmt v0.0.0-20190221155625-df39d6c2d992
+	github.com/smartystreets/goconvey v1.6.4 // indirect
 	github.com/spaolacci/murmur3 v1.1.0
-	github.com/whyrusleeping/cbor-gen v0.0.0-20190910031516-c1cbffdb01bb
-	golang.org/x/xerrors v0.0.0-20190717185122-a985d3407aa7
+	github.com/warpfork/go-wish v0.0.0-20200122115046-b9ea61034e4a // indirect
```
Comment (anorth) [inline go.mod]: Remove these third-party imports if not needed.

File: hamt_test.go [hunk 4]
```diff
@@ -10,9 +10,32 @@ import (
 	"testing"
 	"time"
 
+	cid "github.com/ipfs/go-cid"
+	block "github.com/ipfs/go-block-format"
 	cbor "github.com/ipfs/go-ipld-cbor"
 )
 
+type mockBlocks struct {
+	data map[cid.Cid][]byte
```
Comment (anorth) [inline hamt_test.go]: You might as well `map[cid.Cid]block.Block`, assuming they're immutable

File: gen/main.go [hunk 5]
```diff
@@ -1,32 +1,15 @@
...
 func main() {
-	fi, err := os.Create("cbor_gen.go")
-	if err != nil {
-		panic(err)
-	}
-	defer fi.Close()
-
-	if err := cbg.PrintHeaderAndUtilityMethods(fi, "hamt"); err != nil {
+	// FIXME this will not generate the correct code, leave the cbor_gen.go file untouched.
+	if err := cbg.WriteMapEncodersToFile("cbor_gen.go", "hamt", hamt.Node{}, hamt.KV{}); err != nil {
```
Comment (whyrusleeping) [inline gen/main.go]: wrong call, use `WriteTupleEncodersToFile`

**Pull request merged**